File: src/main/java/com/ject/studytrip/stamp/infra/jpa/StampJpaRepository.java
package com.ject.studytrip.stamp.infra.jpa;

import com.ject.studytrip.stamp.domain.model.Stamp;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface StampJpaRepository extends JpaRepository<Stamp, Long> {
    List<Stamp> findAllByIdIn(List<Long> ids);

    List<Stamp> findAllByTripIdAndDeletedAtIsNull(Long tripId);

    List<Stamp> findAllByTripIdOrderByDeadlineAsc(Long tripId);
}

File: src/main/kotlin/com/ject/studytrip/stamp/infra/jpa/StampJpaRepository.kt
package com.ject.studytrip.stamp.infra.jpa

import com.ject.studytrip.stamp.domain.model.Stamp
import org.springframework.data.jpa.repository.JpaRepository

interface StampJpaRepository : JpaRepository<Stamp, Long> {
    fun findAllByIdIn(stampIds: List<Long>): List<Stamp>

    fun findAllByTripIdAndDeletedAtIsNull(tripId: Long): List<Stamp>

    fun findAllByTripIdOrderByCreatedAtAsc(tripId: Long): List<Stamp>
}
